backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/UserController.java

```diff
@@ -28,7 +30,23 @@ public UserController(UserService service) {
    */
   @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
   @PreAuthorize("isAuthenticated()")
-  public List<User> getUsers(@AuthenticationPrincipal OidcUser oidcUser) {
-    return service.getUsers(oidcUser);
+  public List<User> getUsers(
+      @AuthenticationPrincipal OidcUser oidcUser,
+      @RequestParam(value = "q") Optional<String> searchStr) {
+
+    var users = service.getUsers(oidcUser);
+    if (searchStr.isPresent() && !searchStr.get().isBlank()) {
+      String search = searchStr.get().toLowerCase();
+      return users.stream()
+          .filter(
+              user ->
+                  (user.email() != null && user.email().startsWith(search))
```

Comment:
Die Akzeptanzkriterien besagen: "Es soll die Suche nach E-Mail-Adresse erfolgen, falls Name nicht verfügbar ist". Im Code werden E-Mail, Abkürzung und Name mit einem logischen OR verknüpft. Das bedeutet, es wird gleichzeitig in allen drei Feldern gesucht. Es gibt keine Priorisierung, bei der die E-Mail-Suche nur als Fallback dient.

Außerdem verwendet der code startsWith(), was nur Ergebnisse liefert, die mit dem Suchbegriff beginnen. Die Anforderung ist jedoch die Suche nach dem vollständigen Namen. Ist hier vllt ein contains() besser geeignet, damit nach Vor ODER Nachnamen suchen kann?